### PR TITLE
Fixes and update to pass Sfepy tests

### PR DESCRIPTION
**Pull request by vlukes**
We would like use meshio in our SfePy project [1]. To pass some of our mesh reading/writing tests I have made several updates and fixes. We need to keep information about point and cell groups (subdomains, boundaries, etc.) when reading and writing various formats.

[1] http://sfepy.org

### CODE REVIEW

**Comment by nschloe** [comment]
Hi Vladimir,

thanks for a great PR! The linter isn't happy yet; you can just run `make format` in meshio for the necessary changes.

File: meshio/abaqus/_abaqus.py [hunk 1]
```diff
@@ -245,17 +247,17 @@ def _read_set(f, params_map):
...
-        if line.startswith("*"):
+        if not line or line.startswith("*"):
             break
         if line.strip() == "":
             continue
 
         set_ids += [int(k) for k in line.strip().strip(",").split(",")]
 
-    if "generate" in params_map:
+    if "GENERATE" in params_map:
```
Comment (nschloe) [inline meshio/abaqus/_abaqus.py]: Is this always capitalized? If not, we could check for either or the two. Don't know the abaqus spec by heart.

File: meshio/exodus/_exodus.py [hunk 2]
```diff
@@ -141,8 +141,8 @@ def read(filename):  # noqa: C901
         # elif key == "eb_names":
         #     value.set_auto_mask(False)
         #     eb_names = [b"".join(c).decode("UTF-8") for c in value[:]]
-        elif key == "node_ns":
-            ns = value
+        elif key.startswith("node_ns"):
```
Comment (nschloe) [inline meshio/exodus/_exodus.py]: Out of interest: Why `startswith`?

File: meshio/abaqus/_abaqus.py (hunk 1 above)
Comment (vlukes) [reply meshio/abaqus/_abaqus.py]: You are right, it does not have to be capitalized. So I added `upper()` when parsing the arguments.

File: meshio/exodus/_exodus.py (hunk 2 above)
Comment (vlukes) [reply meshio/exodus/_exodus.py]: It is because the keys for node sets are `node_ns1`, `node_ns2`, etc.

Comment (nschloe) [reply meshio/exodus/_exodus.py]: Best add a comment about that.

Comment (vlukes) [reply meshio/exodus/_exodus.py]: OK, the comment added.

**Comment by tianyikillua** [comment]
For Abaqus, your current implementation of `cell_sets` only store the raw Abaqus cell id's...which is useless outside the INP file (since these point id's / cell id's are not exposed in the `Mesh` object). These id's need to be converted to cell ordering as is in `mesh.cells`. With the new cells / cell_data convention, if we have two types of cells
```
mesh.cells = [("triangle", ...), ("quad", ...)]
```
and a particular cell set `A` is composed of the 1st triangle element, as well as the 2nd and the 6th quad elements, we should have
```
mesh.cell_sets = [(0), (1, 5)]
```

**Comment by vlukes** [comment]
I agree with @tianyikillua, this implementation is only for keeping `cell_sets` when converting via INP format. It should be improved. @tianyikillua, do you have any such Abaqus INP file that contains triangle and quad (or tetra and hex) elements and various cell sets?

**Comment by tianyikillua** [comment]
> I agree with @tianyikillua, this implementation is only for keeping `cell_sets` when converting via INP format. It should be improved. @tianyikillua, do you have any such Abaqus INP file that contains triangle and quad (or tetra and hex) elements and various cell sets?

@vlukes 

Already fixed in https://github.com/nschloe/meshio/pull/666 :-)

I've added a test mesh (`nle1xf3c.inp`) containing two sets of triangles.

You can find lots of example files [here](http://dsk.ippt.pan.pl/docs/abaqus/v6.13/books/bmk/default.htm?startat=ch01s19ach135.html).

**Pull request merged**